src/frequenz/sdk/config/_manager.py

```diff
@@ -175,44 +207,49 @@ def new_receiver(
         Args:
             key: The configuration key to be read by the receiver. If a sequence of
                 strings is used, it is used as a sub-key.
+            config_class: The class object to use to instantiate a configuration. The
+                configuration will be validated against this type too using
+                [`marshmallow_dataclass`][].
             skip_unchanged: Whether to skip sending the configuration if it hasn't
                 changed compared to the last one received.
+            base_schema: An optional class to be used as a base schema for the
+                configuration class. This allow using custom fields for example. Will be
+                passed to [`marshmallow_dataclass.class_schema`][].
+            marshmallow_load_kwargs: Additional arguments to be passed to
+                [`marshmallow.Schema.load`][].
 
         Returns:
             The receiver for the configuration.
         """
-        recv_name = key if isinstance(key, str) else ":".join(key)
-        receiver = self.config_channel.new_receiver(name=recv_name, limit=1)
-
-        def _get_key_or_error(
-            config: Mapping[str, Any]
-        ) -> Mapping[str, Any] | InvalidValueForKeyError | None:
-            try:
-                return _get_key(config, key)
-            except InvalidValueForKeyError as error:
-                return error
-
-        key_receiver = receiver.map(_get_key_or_error)
+        receiver = self.config_channel.new_receiver(name=f"{self}:{key}", limit=1).map(
+            lambda config: _load_config_with_logging_and_errors(
+                config,
+                config_class,
+                key=key,
+                base_schema=base_schema,
+                marshmallow_load_kwargs=marshmallow_load_kwargs,
+            )
+        )
```

Comment:
The same, please add extra filter `IsNotException`, then this part will be completely seprate from others and you won't have to change  `_not_equal_with_logging` :)

Reply:
I don't fully understand your suggestion here, can you maybe post a small diff with the suggested changes?

Just for context, I tried not to apply many filters on purpose, to avoid the extra overhead, but if it really makes the code cleaner, I guess we can afford it anyways, config updates should be pretty seldom.

Reply:
here :) 
https://github.com/frequenz-floss/frequenz-sdk-python/pull/1134#discussion_r1891424080